### PR TITLE
feat(glm5next): expose auxiliary states for DFlash2

### PR DESCRIPTION
**Pull request by MadeBy561**
## Summary

- expose requested GLM-5.3 target-layer features through the existing `SupportsEagle3` auxiliary-state interface
- materialize deferred mHC state with `hc_post` and contract its residual streams before capture
- gather final and auxiliary states in one sequence-parallel collective
- preserve the existing fast forward loop when no external drafter requests features

## Why

Jovian Judgement already contains the DFlash2 proposer, candidate selector, probabilistic drafting, rejection sampling, and non-causal draft attention. The missing piece for `incoai/GLM-5.3-Flash-DFlash2` was the GLM target hook that returns the five features the checkpoint was trained against (zero-based layers 5, 14, 24, 33, and 42).

GLM defers its mHC reconstruction, so capturing the raw loop tensor is not the trained target feature. This change reconstructs it with the layer-owned mHC operators before returning it through the existing auxiliary-state contract.

Duplicate check: no open GLM-5.3 DFlash or auxiliary-hidden-state PR targets `dev/jovian-judgement`.

## Validation

```text
python -m pytest -q tests/models/test_glm5next_model.py \
  -k "external_drafters or materializes_deferred_mhc or dflash_layer_boundaries"
3 passed, 25 deselected

ruff 0.14.0 check vllm/models/glm5next/nvidia/model.py tests/models/test_glm5next_model.py
All checks passed

ruff 0.14.0 format --check vllm/models/glm5next/nvidia/model.py tests/models/test_glm5next_model.py
2 files already formatted
```

Live TP4 SM120 qualification used normal model-card sampling (`temperature=1`, `top_p=0.95`) with seven draft tokens per K8 verification step:

- 59 verification steps / 413 proposed draft tokens
- 271 accepted draft tokens (65.6%)
- 4.59 accepted draft tokens per step, 5.59 tokens per verification including the target token
- seventh draft position accepted on 26/59 steps (44.1%)
- sustained single-stream code generation around 275-287 tok/s, peaking at 290 tok/s
- completed an Estonia C10/N60 run at 57/60, versus 56/60 for the matched MTP3 control; this is not presented as an accuracy gain, only as evidence against a DFlash-specific collapse

The target remained native B12X sparse MLA + NVFP4/A4 MoE + B12X PCIe all-reduce with FP8 KV.

## Scope

This does not change target weights, quantization, sampling, or the no-drafter path. External multimodal draft embeddings remain out of scope.

AI assistance was used for implementation, testing, and documentation; the commit includes an `Assisted-by` trailer.

## Summary by CodeRabbit

- **New Features**
  - Added Eagle3 speculative decoding support for GLM-5-Next language and conditional-generation models.
  - Enabled collection and handling of auxiliary hidden states during model execution.

- **Tests**
  - Added coverage confirming Eagle3 support.
  - Added validation for deferred auxiliary hidden-state materialization and aggregation.

### CODE REVIEW

File: vllm/models/glm5next/nvidia/model.py [hunk 1]
```diff
@@ -957,7 +1001,9 @@ def load_weights(self, weights: Iterable[tuple[str, torch.Tensor]]) -> set[str]:
         return loaded_params
 
 
-class Glm5NextForCausalLM(nn.Module, HasInnerState, MixtureOfExperts, IsHybrid):
+class Glm5NextForCausalLM(
+    nn.Module, HasInnerState, MixtureOfExperts, IsHybrid, SupportsEagle3
+):
```
[the remark this answers is not in the document]
Comment (MadeBy561) [reply vllm/models/glm5next/nvidia/model.py:1024]: Fixed in 577e4155d. The fallback uses `(6, 15, 25, 34, 43)`, not the checkpoint’s raw `(5, 14, 24, 33, 42)`: DFlash `target_layer_ids` are zero-based decoder-layer indices, while vLLM converts them to one-based layer-boundary indices before calling `set_aux_hidden_state_layers` (`i + 1` in `eagle3_utils.py` and `gpu_model_runner.py`). The new regression test covers both GLM wrappers and the helper now has the requested Google-style docstring.

**Comment by MadeBy561** [comment]
Already handled by Luke closing
